sdl_android/src/main/java/com/smartdevicelink/streaming/RTPH264Packetizer.java

```diff
@@ -0,0 +1,488 @@
+		onEncoderOutput(mNALUnitReader, ptsInUs);
+	}
+
+	private void onEncoderOutput(NALUnitReader nalUnitReader, long ptsInUs) {
+		if (mPaused) {
+			return;
+		}
+
+		ByteBuffer nalu;
+
+		while ((nalu = nalUnitReader.getNalUnit()) != null) {
+			if (mWaitForIDR) {
+				if (isIDR(nalu)) {
+					mWaitForIDR = false;
+				} else {
+					continue;
+				}
+			}
+			outputRTPFrames(nalu, ptsInUs, nalUnitReader.hasConsumedAll());
+		}
+	}
+
+	private boolean outputRTPFrames(ByteBuffer nalu, long ptsInUs, boolean isLast) {
+		if (RTP_HEADER_LEN + nalu.remaining() > MAX_RTP_PACKET_SIZE) {
+			// Split into multiple Fragmentation Units ([5.8] in RFC 6184)
+			byte firstByte = nalu.get();
+			boolean firstFragment = true;
+			boolean lastFragment = false;
+
+			while (nalu.remaining() > 0) {
+				int payloadLength = MAX_RTP_PACKET_SIZE - (RTP_HEADER_LEN + FU_INDICATOR_LEN + FU_HEADER_LEN);
+				if (nalu.remaining() <= payloadLength) {
+					payloadLength = nalu.remaining();
+					lastFragment = true;
+				}
+
+				ByteBuffer frame = allocateRTPFrame(FU_INDICATOR_LEN + FU_HEADER_LEN + payloadLength,
+				                                    false, isLast, ptsInUs);
+				// FU indicator
+				frame.put((byte)((firstByte & 0xE0) | TYPE_FU_A));
+				// FU header
+				frame.put((byte)((firstFragment ? 0x80 : lastFragment ? 0x40 : 0) | (firstByte & 0x1F)));
+				// FU payload
+				frame.put(nalu.array(), nalu.position(), payloadLength);
+				nalu.position(nalu.position() + payloadLength);
+				frame.flip();
+
+				try {
+					mOutputQueue.put(frame);
+				} catch (InterruptedException e) {
+					Thread.currentThread().interrupt();
+					return false;
+				}
+
+				firstFragment = false;
+			}
+		} else {
+			// Use Single NAL Unit Packet ([5.6] in RFC 6184)
+			ByteBuffer frame = allocateRTPFrame(nalu.remaining(), false, isLast, ptsInUs);
+			frame.put(nalu);
+			frame.flip();
+
+			try {
+				mOutputQueue.put(frame);
+			} catch (InterruptedException e) {
+				Thread.currentThread().interrupt();
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	private ByteBuffer allocateRTPFrame(int rtpPayloadLen,
+	                                    boolean hasPadding, boolean isLast, long ptsInUs) {
+		assert rtpPayloadLen > 0;
+		assert ptsInUs > 0;
+
+		int packetLength = RTP_HEADER_LEN + rtpPayloadLen;
+		assert packetLength <= MAX_RTP_PACKET_SIZE;
+		int ptsIn90kHz = (int)(ptsInUs * 9 / 100) + mInitialPTS;
+
+		ByteBuffer frame = ByteBuffer.allocate(FRAME_LENGTH_LEN + packetLength);
+		frame.order(ByteOrder.BIG_ENDIAN);
+		frame.putShort((short)packetLength);
+
+		// Version = 2, Padding = hasPadding, Extension = 0, CSRC count = 0
+		frame.put((byte)(0x80 | (hasPadding ? 0x20 : 0)))
+			// Marker = isLast, Payload type = mPayloadType
+			.put((byte)((isLast ? 0x80 : 0) | (mPayloadType & 0x7F)))
+			.putChar(mSequenceNum)
+			.putInt(ptsIn90kHz)
+			.putInt(mSSRC);
+
+		assert frame.position() == FRAME_LENGTH_LEN + RTP_HEADER_LEN;
```

Comment:
I don't believe `assert` should not be used in production code. It should be replaced with if statements and thrown exceptions.

Reply:
OK, replaced with if() check and RuntimeException, in commit https://github.com/smartdevicelink/sdl_android/pull/592/commits/9663bc8ba110fc647bdfb54dfc53435fca29f366.